packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -105,14 +106,33 @@ void main() {
         dirToDelete.deleteSync(recursive: true);
       }
     }
+    Timer timeout = new Timer(_kPlatformTimeout, () {
+      process.stderr
+        .transform(UTF8.decoder)
+        .transform(const LineSplitter())
+        .listen((String line) {
+          if (line != null)
+            print(line);
+        });
+      finalize();
+      throw new TimeoutException("Test took too long", _kPlatformTimeout);
+    });
+    void cancelTimerAndFinalize() {
+      finalize();
+      if (timeout != null) {
+        Timer timerToCancel = timeout;
+        timeout = null;
+        timerToCancel.cancel();
```

Comment:
that's an odd way to do it. Why not (instead of the whole if block):
```dart
   timeout?.cancel();
   timeout = null;
```
If there's a reason to do it this way, please put a comment here explaining why.